### PR TITLE
Fix WheelTimer implementation that can expired timeout early

### PR DESCRIPTION
**Pull request by udovichenkoAlexander**
When entries insert in the end of timer queue, then unnecessary entry inserted (with duplicated key).
This can lead to some timeouts expired early and consume memory.

This can be checked with following test ([to this file](https://github.com/element-hq/synapse/commits/9512b84a727a6519a407c72764b4c5ec36323a78/tests/util/test_wheel_timer.py))
```py
def test_multi_insert_then_past(self) -> None:
    wheel: WheelTimer[object] = WheelTimer(bucket_size=5)

    obj1 = object()
    obj2 = object()
    obj3 = object()
    wheel.insert(100, obj1, 150)
    wheel.insert(100, obj2, 160)
    wheel.insert(100, obj3, 155)
    
    self.assertListEqual(wheel.fetch(110), [])
    self.assertListEqual(wheel.fetch(158), [obj1])
```
Look at the entries after obj2 insert:
```
_Entry(end_key=21, elements=set())
_Entry(end_key=22, elements=set())
_Entry(end_key=23, elements=set())
_Entry(end_key=24, elements=set())
_Entry(end_key=25, elements=set())
_Entry(end_key=26, elements=set())
_Entry(end_key=27, elements=set())
_Entry(end_key=28, elements=set())
_Entry(end_key=29, elements=set())
_Entry(end_key=30, elements=set())
_Entry(end_key=31, elements={<object object at 0x7a129fb7ab80>})
_Entry(end_key=31, elements=set()) <-- this entry has duplicate key
_Entry(end_key=32, elements=set())
_Entry(end_key=33, elements={<object object at 0x7a129fb7ab70>})
```
gaps and duplicates cause errors when search Entry to insert in the middle of queue:
`self.entries[max(min_key, then_key) - min_key].elements.add(obj)`

This Pull request fix this with some minor refactoring (remove `current_tick` unused variable) and add this test case to unit tests.

### CODE REVIEW

**Review by MadLittleMods — CHANGES_REQUESTED** [review]
Nice catch! Makes sense after going through the motions

Strictly just needs the changelog fix. Updating tests also seems like a good change but we can always do that in a follow-up PR.

**Comment by udovichenkoAlexander** [comment]
I fixed changelog and refactored wheel tests to use strings intead of objects

File: changelog.d/17850.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+Fix bug when some presence and typing timeouts can expire early.
```
Comment (MadLittleMods) [inline changelog.d/17850.bugfix:1]: @udovichenkoAlexander [Run the linters](https://element-hq.github.io/synapse/latest/development/contributing_guide.html#run-the-linters) to fix up the [failures showing up in CI](https://github.com/element-hq/synapse/actions/runs/11630429910/job/32397409108?pr=17850)

Comment (MadLittleMods) [reply changelog.d/17850.bugfix:1]: @udovichenkoAlexander Linting still failing -> https://github.com/element-hq/synapse/actions/runs/11669623620/job/32492010646?pr=17850

Comment (udovichenkoAlexander) [reply changelog.d/17850.bugfix:1]: I run linter locally, it's OK now

**Review by MadLittleMods — APPROVED** [review]

**Comment by MadLittleMods** [comment]
Thanks for the contribution @udovichenkoAlexander 🐄

**Pull request merged**